src/renderer/frames/cluster-frame/cluster-frame-layout-child-component.injectable.tsx

```diff
@@ -9,51 +9,62 @@ import { MainLayout } from "../../components/layout/main-layout";
 import { Sidebar } from "../../components/layout/sidebar";
 import { Dock } from "../../components/dock";
 import styles from "./cluster-frame.module.css";
+import type { IComputedValue } from "mobx";
 import { computed } from "mobx";
 import currentRouteComponentInjectable from "../../routes/current-route-component.injectable";
 import { Redirect } from "react-router";
 import startUrlInjectable from "./start-url.injectable";
 import currentPathInjectable from "../../routes/current-path.injectable";
 import { observer } from "mobx-react";
+import { withInjectables } from "@ogre-tools/injectable-react";
+
+interface Dependencies {
+  currentRouteComponent: IComputedValue<React.ElementType<any> | undefined>;
+  startUrl: IComputedValue<string>;
+  currentPath: IComputedValue<string>;
+}
+
+const NonInjectedClusterFrameLayout = observer((props: Dependencies) => {
+  const Component = props.currentRouteComponent.get();
+  const starting = props.startUrl.get();
+  const current = props.currentPath.get();
+
+  return (
+    <MainLayout sidebar={<Sidebar />} footer={<Dock />}>
+      {Component ? (
+        <Component />
+      ) : // NOTE: this check is to prevent an infinite loop
+        starting !== current ? (
+          <Redirect to={starting} />
+        ) : (
+          <div className={styles.centering}>
+            <div className="error">
+              An error has occured. No route can be found matching the
+              current route, which is also the starting route.
+            </div>
+          </div>
+        )}
+    </MainLayout>
+  );
+});
+
+const ClusterFrameLayout = withInjectables<Dependencies>(NonInjectedClusterFrameLayout, {
+  getProps: (di, props) => ({
+    ...props,
+    currentRouteComponent: di.inject(currentRouteComponentInjectable),
+    startUrl: di.inject(startUrlInjectable),
+    currentPath: di.inject(currentPathInjectable),
```

Comment:
Ok, so this fixes the injection cycle by deferring injection to only until the component is rendered, which is correct. We remain a bit baffled why the cycle happened only now, and not long before. Curious.

Reply:
I think this should be merged as well.

Reply:
I am especially curious because the error states a cycle with an edge that didn't actually exist.

Reply:
@Nokel81 are we going to split this into a separate PR?

Reply:
Sure, I can make this a separate PR

Reply:
> I am especially curious because the error states a cycle with an edge that didn't actually exist.

This is true, and we now have some understanding about why this is, and the solution is in the works.